service/key/key.go

```diff
@@ -38,40 +38,10 @@ func ConfigMapName(customObject kvmtpr.CustomObject, node spec.Node, prefix stri
 	return fmt.Sprintf("%s-%s-%s", prefix, ClusterID(customObject), node.ID)
 }
 
-func ConfigMapNames(customObject kvmtpr.CustomObject) []string {
-	var names []string
-
-	for _, node := range customObject.Spec.Cluster.Masters {
-		name := ConfigMapName(customObject, node, MasterID)
-		names = append(names, name)
-	}
-
-	for _, node := range customObject.Spec.Cluster.Workers {
-		name := ConfigMapName(customObject, node, WorkerID)
-		names = append(names, name)
-	}
-
-	return names
-}
-
 func DeploymentName(prefix string, nodeID string) string {
 	return fmt.Sprintf("%s-%s", prefix, nodeID)
 }
 
-func DeploymentNames(customObject kvmtpr.CustomObject) []string {
-	var names []string
-
-	for _, masterNode := range customObject.Spec.Cluster.Masters {
-		names = append(names, DeploymentName(MasterID, masterNode.ID))
-	}
-
-	for _, workerNode := range customObject.Spec.Cluster.Workers {
-		names = append(names, DeploymentName(WorkerID, workerNode.ID))
-	}
-
-	return names
-}
```

Comment:
We have been using a deterministic list of config maps and deployments to fetch the current resources in the Kubernetes API. This does not work in case we scale down because the resources we want to fetch as current state could not be found because the knowledge of making up the name we were looking for has been gone by updating the custom object beforehand. The solution is to list all resources within the cluster namespace. Also see the code below within the resource implementation.